src/utilities/bcl/RemoteBCL.cpp

```diff
@@ -763,36 +798,22 @@ bool RemoteBCL::startComponentLibrarySearch(const std::string& searchTerm, const
                        }
                      });
 
-  //LOG(Warn, remoteUrl() + toString(builder.to_string()));
+  // LOG(Debug, m_remoteUrl << builder.to_string());
 
   return true;
 }
 
-bool RemoteBCL::waitForLock(int msec) const {
-  int msecPerLoop = 20;
-  int numTries = msec / msecPerLoop;
-  int current = 0;
-  while (true) {
-    // if no request was made and the optional is empty return
-    if (!m_httpResponse) {
-      return false;
-    }
-
-    // if we can get the lock then the download is complete
-    if (m_httpResponse->is_done()) {
-      return true;
-    }
-
-    // DLM: no longer calls Application::instance().processEvents(msecPerLoop);
-    // this calls process events
-    System::msleep(msecPerLoop);
+bool RemoteBCL::waitForLock() const {
 
-    if (current > numTries) {
-      LOG(Error, "waitForLock timeout");
-      break;
-    }
+  if (!m_httpResponse) {
+    return false;
+  }
 
-    ++current;
+  try {
+    m_httpResponse->wait();
+    return true;
+  } catch (const std::exception& e) {
+    LOG(Error, "Request to url '" << m_remoteUrl << " 'failed with message: " << e.what());
```

Comment:
Fix the crash portion I found while investigating #4014 and trying twice to download